File: project-euler/c/7.c
#include <stdio.h>

// Finds the next prime number.
int next_prime(int num)
{
	do
	{
		int prime = 1;
		num++;
		for (int i = 2; i < num; i++)
			if (num % i == 0) { prime = 0; break; }
		if (prime) return num;
	} while(1);
}

int main()
{
	int prime = 1;
	for (int i = 1; i <= 10001; i++)
		prime = next_prime(prime);
	printf("%d", prime);
	return 0;
}

File: challenges/project-euler/python3/7.py
# is_prime() and next_prime() borrowed from 3.py

# Check if a number is prime or not.
def is_prime(num):
    for i in range(2, (num // 2) + 1):
        if (num % i == 0): return False
    return True

# Get the next prime number, after a given number.
def next_prime(num):
    while True:
        num += 1
        if is_prime(num): return num

# Gets the nth prime number:
def nth_prime(num):
    current_prime = 1
    for i in range(0, num):
        current_prime = next_prime(current_prime)
    return current_prime

print(nth_prime(10001))
